main.tf

```diff
@@ -121,6 +121,8 @@ resource "google_storage_bucket" "buckets" {
         created_before             = lookup(lifecycle_rule.value.condition, "created_before", null)
         with_state                 = lookup(lifecycle_rule.value.condition, "with_state", lookup(lifecycle_rule.value.condition, "is_live", false) ? "LIVE" : null)
         matches_storage_class      = contains(keys(lifecycle_rule.value.condition), "matches_storage_class") ? split(",", lifecycle_rule.value.condition["matches_storage_class"]) : null
+        matches_prefix             = lookup(lifecycle_rule.value.condition, "matches_prefix", null)
+        matches_suffix             = lookup(lifecycle_rule.value.condition, "matches_suffix", null)
```

Comment:
Since `condition` is a `map(string)` and there fields need a [list input per provider](https://github.com/hashicorp/terraform-provider-google/blob/d2ea6580ef848edd94862d13ab7cddcb162df904/google/resource_storage_bucket.go#L213), we would need to follow an approach like `matches_storage_class` taking these in as a comma delimited string.